common/unit_test/Test_Padding.cpp

```diff
@@ -43,434 +65,535 @@ void test_reshape1D_1DView(bool is_C2R) {
   EXPECT_TRUE(shape_crop == ref_shape_crop);
 }
 
-void test_reshape1D_2DView(bool is_C2R) {
+template <typename T>
+void test_reshape1D_2DView() {
   const int n0 = 30, n1 = 15;
-  View2D<double> x("x", n0, n1);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+
+  View2D<T> x_out("x_out", n0, n1);
   constexpr int DIM = 2;
 
   shape_type<2> default_shape({n0, n1});
-
   for (int axis0 = 0; axis0 < DIM; axis0++) {
+    shape_type<2> in_shape = default_shape;
+    in_shape.at(axis0)     = get_c2r_shape(x_out.extent(axis0), is_C2R);
+    auto [_n0, _n1]        = in_shape;
+    View2D<Kokkos::complex<double>> x_in("x_in", _n0, _n1);
     for (int i0 = -1; i0 <= 1; i0++) {
       shape_type<2> ref_shape = default_shape;
-      std::size_t n_new       = static_cast<std::size_t>(x.extent(axis0) + i0);
-      ref_shape.at(axis0)     = get_c2r_shape(n_new, is_C2R);
-      auto modified_shape     = KokkosFFT::Impl::get_modified_shape(
-          x, shape_type<1>{n_new}, axes_type<1>{axis0}, is_C2R);
+      std::size_t n_new   = static_cast<std::size_t>(x_in.extent(axis0) + i0);
+      ref_shape.at(axis0) = get_c2r_shape(n_new, is_C2R);
+
+      auto modified_shape = KokkosFFT::Impl::get_modified_shape(
+          x_in, x_out, shape_type<1>{n_new}, axes_type<1>{axis0});
 
       EXPECT_TRUE(modified_shape == ref_shape);
     }
   }
 }
 
-void test_reshape1D_3DView(bool is_C2R) {
+template <typename T>
+void test_reshape1D_3DView() {
   const int n0 = 30, n1 = 15, n2 = 8;
-  View3D<double> x("x", n0, n1, n2);
+  bool is_C2R = !KokkosFFT::Impl::is_complex<T>::value;
+
+  View3D<T> x_out("x_out", n0, n1, n2);
   constexpr int DIM = 3;
 
   shape_type<3> default_shape({n0, n1, n2});
-
   for (int axis0 = 0; axis0 < DIM; axis0++) {
+    shape_type<3> in_shape = default_shape;
+    in_shape.at(axis0)     = get_c2r_shape(x_out.extent(axis0), is_C2R);
+    auto [_n0, _n1, _n2]   = in_shape;
+    View3D<Kokkos::complex<double>> x_in("x_in", _n0, _n1, _n2);
     for (int i0 = -1; i0 <= 1; i0++) {
       shape_type<3> ref_shape = default_shape;
-      std::size_t n_new       = static_cast<std::size_t>(x.extent(axis0) + i0);
-      ref_shape.at(axis0)     = get_c2r_shape(n_new, is_C2R);
-      auto modified_shape     = KokkosFFT::Impl::get_modified_shape(
-          x, shape_type<1>{n_new}, axes_type<1>{axis0}, is_C2R);
+      std::size_t n_new   = static_cast<std::size_t>(x_in.extent(axis0) + i0);
```

Comment:
```suggestion
      auto n_new = x_in.extent(axis0) + i0;
```

`static_cast<size_t>` can be done later if needed.